core/coreapi/object.go

```diff
@@ -0,0 +1,190 @@
+package coreapi
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"io/ioutil"
+
+	"github.com/ipfs/go-ipfs/merkledag/utils"
+
+	coreiface "github.com/ipfs/go-ipfs/core/coreapi/interface"
+	dag "github.com/ipfs/go-ipfs/merkledag"
+	ft "github.com/ipfs/go-ipfs/unixfs"
+)
+
+type ObjectAPI CoreAPI
+
+func (api *ObjectAPI) New(ctx context.Context) (coreiface.Node, error) {
+	node := new(dag.ProtoNode)
```

Comment:
On the CLI/API, `object new` takes a template argument which can be `unixfs-dir` for a directory. We can do this here similarly to the `KeyAPI.WithType()` option in #4477.